packages/react-native-reanimated/src/createAnimatedComponent/NativeEventsManager.ts

```diff
@@ -77,23 +77,52 @@ export class NativeEventsManager implements INativeEventsManager {
     });
   }
 
-  private getEventViewTag() {
+  private getEventViewTag(componentUpdate: boolean = false) {
     // Get the tag for registering events - since the event emitting view can be nested inside the main component
     const componentAnimatedRef = this.#managedComponent
-      ._component as AnimatedComponentRef;
-    let newTag: number;
+      ._componentRef as AnimatedComponentRef & {
+      __nativeTag?: number;
+      _nativeTag?: number;
+    };
     if (componentAnimatedRef.getScrollableNode) {
+      /*
+        In most cases, getScrollableNode() returns a view tag, and findNodeHandle is not required. 
+        However, to cover more exotic list cases, we will continue to use findNodeHandle 
+        for consistency. For numerical values, findNodeHandle should return the value immediately, 
+        as documented here: https://github.com/facebook/react/blob/91061073d57783c061889ac6720ef1ab7f0c2149/packages/react-native-renderer/src/ReactNativePublicCompat.js#L113
+      */
       const scrollableNode = componentAnimatedRef.getScrollableNode();
-      newTag = findNodeHandle(scrollableNode) ?? -1;
-    } else {
-      newTag =
-        findNodeHandle(
-          this.#componentOptions?.setNativeProps
-            ? this.#managedComponent
-            : componentAnimatedRef
-        ) ?? -1;
+      if (typeof scrollableNode === 'number') {
```

Comment:
Does this also work as intended on Fabric where there are shadow node wrappers instead of view tags (numbers)?

Reply:
On both platforms, `getScrollableNode` returns a number - I've checked it for both ScrollView and FlatList.